pkg/istio-agent/xds-proxy.go

```diff
@@ -87,7 +87,9 @@ func initXdsProxy(ia *Agent) (*XdsProxy, error) {
 	}
 
 	go func() {
-		_ = proxy.downstreamGrpcServer.Serve(proxy.downstreamListener)
+		if err := proxy.downstreamGrpcServer.Serve(proxy.downstreamListener); err != nil {
```

Comment:
@rshriram I verified it again and Serve can return err and this what we do in istiod code as well https://github.com/istio/istio/blob/master/pilot/pkg/bootstrap/server.go#L366

Reply:
In Istiod we do Errorf not fatalf. Is this one of those things that will return an err when we shutdown and this will cause it to force exit instead of graceful exit?

Reply:
The serve can return non nil err when listener fails to accept a new connection (in this case Envoy) for some unknown reason https://github.com/grpc/grpc-go/blob/44d73dff99bfd2aff2cb587bb942d6f8bf8148f7/server.go#L679 - not related to stop/graceful stop. 
Examples in grpc-go also do this https://github.com/grpc/grpc-go/blob/master/examples/helloworld/greeter_server/main.go#L48

I will change this to Errorf to match with what we do in Istiod.